### ISSUE TITLE
Plugin APScheduler: Database backend

### ISSUE THREAD

**Issue by mattrasband** [pyslackers/sirbot-plugins#9]
Auto post a message to a specific channel on a recurring schedule.

Use Case:

* Weekly "Who's Hiring" thread
* Weekly "Who's Looking" thread

**Comment by ovv** [comment]
[APScheduler](https://apscheduler.readthedocs.io/en/latest/index.html) seems a good library to implement that

**Comment by mattrasband** [comment]
Yeah, I think that APScheduler is probably the best option - not sure the best way to integrate it with the bot framework. I guess we could pass in a scheduler instance to the plugin and they use that directly (or abstract it out and have a custom plugin type?)

**Comment by ovv** [comment]
I think it's best to make a custom plugin out of it, can of like the DB plugin so the core stay lighter. 

Everyone might not need a scheduler and I don't really see a downside of doing it that way.

**Comment by ovv** [comment]
Still wondering if we need to have an optional database store for the job.

At the moment all jobs are recreated at startup so it's not needed but maybe it would be usefull
